src/components/DappBrowser/Homepage.tsx

```diff
@@ -144,14 +145,30 @@ const Card = React.memo(function Card({
   goToUrl,
   site,
   showMenuButton,
+  index,
 }: {
   goToUrl: (url: string) => void;
   showMenuButton?: boolean;
   site: Omit<Site, 'timestamp'>;
+  index?: number;
 }) {
   const { isDarkMode } = useColorMode();
+  const browserHistoryStore = useBrowserHistoryStore();
```

Comment:
this subscribes to the entire browserHistoryStore. so if ANYTHING in the store changes, this component will rerender. what you should do is `const hasVisited = useBrowserHistoryStore(state => state.hasVisited);`. this way, the component will only be subscribed to that aspect of the store